src/methods.jl

```diff
@@ -179,3 +179,71 @@ function trimview(frame::AbstractArray, idxs)
 
     return selectdim(frame, d, complement_idxs)
 end
+
+
+"""
+    crop(frame::AbstractArray, shape; force_equal = true)
+
+Crops the image to the shape specified asumming center as the anchor.
+
+Shape speficies the size of the cropped array, force_equal when set to true forces the cropped rows and columns to be equal on both sides
+otherwise the cropped frame contains one extra column on left and one extra row on top if necessary.
+
+# Examples
+```jldoctest
+julia> frame = reshape(1:25, (5, 5));
+
+julia> crop(frame, (3, 3))
+3×3 Array{Int64,2}:
+ 7  12  17
+ 8  13  18
+ 9  14  19
+
+```
+
+# See Also
+* [`cropview`](@ref)
+"""
+crop(frame::AbstractArray, shape; kwargs...) = copy(cropview(frame, shape; kwargs...))
+
+
+"""
+    cropview(frame::AbstractArray, shape; force_equal = true)
+
+Crops the image to the shape specified asumming center as the anchor.
+
+This function is same as the [`crop`](@ref) function but returns a view of the frame.
+
+!!! note
+    This function returns a view of the frame, so any modification to output
+    array will result in modification of frame.
+
+# See Also
+* [`crop`](@ref)
+"""
+function cropview(frame::AbstractArray, shape; force_equal = true)
+    shape_diff = size(frame) .- shape
+
+    # testing error
+    any(s -> s < 0, shape_diff) && error("crop size ($shape) can't be larger than frame size ($(size(frame)))")
+    any(s -> s < 0, shape) && error("crop size ($shape) cant't be negative")
+
+    # calculating range of cropped frame
+    range = []
+    for i = 1:2
+        strip = size(frame, i)
+        if iseven(shape_diff[i])
+            push!(range, 1 + shape_diff[i] ÷ 2 : strip - shape_diff[i] ÷ 2)
+        else
+            if force_equal
+                @warn "Shape was increased to keep equal trim size"
+                push!(range, 1 + (shape_diff[i] - 1) ÷ 2 : strip - (shape_diff[i] - 1) ÷ 2)
+            else
+                push!(range, 1 + (shape_diff[i] - 1) ÷ 2 : strip - (shape_diff[i] + 1) ÷ 2)
```

Comment:
Oh, right, I didn't notice this is a range.  Pay attention to precedence of operators here, you may not be doing what you expect here.  Use parentheses as appropriate

Reply:
I wanted to divide first, so I think this would work fine! Still, I will put the parentheses.

Reply:
My worry is in particular about `:`, it has a weird precedence.  When I have ranges more complex than `a:b` I prefer to wrap both sides of `:` with parentheses: `(a + b * c) : (d ^ e / f)`

Reply:
Added the parentheses!